caiman/motion_correction.py

```diff
@@ -2497,7 +2497,15 @@ def motion_correction_piecewise(fname, splits, strides, overlaps, add_to_movie=0
         T = shape[0]
         del dataset
     elif extension == '.npy':
-        raise Exception('Numpy not supported at the moment')
+        # assumes the data is saved in a 3x3 array for the moment with order
```

Comment:
@AndrewMicallef we might have a problem here. You should be able to load only part of the movie in memory, otherwise when you perform parallel motion correction every process will load in memory the full movie, therefore occupying a total RAM memory n_processes*original_movie_size_single_precision.

Reply:
I was following the pattern of the other file types here, when I hacked this together it wasn't clear how the data actually get's past to the motion correction algorithm. It seams to me that at this juncture the only used information is the array shape. 
In any case, using np.load(array, memmap_mode='r') prevents the array being loaded into the memory, so I'm not sure you get the explosion you are mentioning. Either way, because the array I use doesn't go anywhere it is a bit of a waste, so I'd probably be better just grabbing the header info from the numpy file...

I'm actually heading on holiday for two weeks, so I won't be hacking a way at this for a little while

Reply:
Memory mapping the npy file is a great idea, when you are back we can chat about it. Thank you!